[IV-23-24] Objetivo 6: Integración continua (CI)

marinajcs opened the pull request:
* [x] ¿Se han establecido criterios *a priori* para elegir el sistema de
integración continua más conveniente y se han calificado los sistemas que
cumplan esos requisitos según esos criterios?
* [x] ¿Se han examinado varios sistemas de integración continua?
* [x] ¿Se han configurado varios sistemas de integración continua?
* [x] ¿Se ha configurado correctamente el *Checks API* en los sistemas en los que
   sea necesario para que aparezca correctamente en GitHub (y se pueda comprobar
   desde los tests)?
* [x] ¿Uno de los sistemas configurados permite comprobar cuales son las versiones
   dellenguaje con las que funciona correctamente nuestra aplicación?
* [x] ¿Se escogen de forma adecuada las versiones del lenguaje que se testean,
   tanto en el sistema de CI como en el contenedor Docker?
* [x] ¿Se han justificado correctamente las versiones del lenguaje que se están
testeando y se ha comprobado que no se comprueban varias veces lo mismo (la
misma versión en CI y en el contenedor Docker en otro sistema CI?

marinajcs committed:
Documentación provisional, closes #37

Documentación incompleta de la elección de CI que se irá completando a medida que se prueben herramientas y se vayan descartando

marinajcs committed:
Comprobación de versiones, closes #39 #

Se añade workflow con GitHub Actions que comprueba las versiones (LTS, actual, anterior...) de Node con las que funciona el proyecto

marinajcs committed:
Prueba con buildkite, closes #40

marinajcs committed:
Probando buildkite por 2º vez, #40

Tras configuración de pipeline en la interfaz de buildkite, se procede a intentarlo de nuevo

marinajcs committed:
Actualiza documentación CI, fixes #37

Se documenta la elección de las herramientas de CI consideradas y finalmente seleccionadas, lo que también se ve reflejado en el README.

marinajcs committed:
Añadidos paths CI en yaml, fixes #37

carlotiii30 reviewed (approved):
Veo que has hecho un gran trabajo de investigación. Lo veo perfecto.

MarioGuisado reviewed (approved):
En general muy completo, aunque quizás en los criterios convendría definir uno que indique que aquellas herramientas que sean usadas por muchos otros alumnos se descartarán, porque es lo que se ha usado para eliminar muchas de las opciones consideradas.

marinajcs committed:
Añadido criterio de frecuencia de uso de herramienta, fixes #37

Dado que se ha usado para descartar muchas de las herramientas listadas, se ha incluido como criterio preestablecido si la herramienta en cuestión ha sido ampliamente utilizada por varios alumnos anteriormente.

Enaraque commented on docs/eleccion-ci.md:
```diff
@@ -0,0 +1,94 @@
+
+Etre las opciones de herramientas de CI, se encuentran:
+
+* [**Jenkins**](https://www.jenkins.io/): herramienta de automatización de código abierto
+para construir, probar y desplegar software. Se terminó descartando
```
Jenkins es una herramienta que se tiene que instalar localmente. No la consideraría como una herramienta que podríamos usar a no ser que desees hacer una instalación local en tu PC

marinajcs commented:
@JJ Listo para revisión

Enaraque reviewed (approved):
En general lo veo bastante completo y compruebas varias versiones. Te he dejado un comentario respecto a una de las herramientas consideradas, pero buen trabajo! 💪

JJ commented on docs/eleccion-ci.md:
```diff
@@ -0,0 +1,94 @@
+* [**CircleCI**](https://circleci.com/): plataforma de CI/CD basada en la nube que permite
+la construcción y prueba automatizadas de proyectos de software. Es
+una opción sólida en general,aunque ya ha sido escogida varias veces.
+* [**Atlassian Bamboo**](https://www.atlassian.com/es/software/bamboo): solución de CI/CD de Atlassian. Requiere
```
Si vas a descartarlas "automáticamente", incluirlas no genera ninguna información. Simplemente establece criterios de búsqueda e incluye solo las que cumplan esos criterios de búsqueda.

marinajcs replied:
Resuelto en https://github.com/marinajcs/asignacionTareas/pull/38/commits/b9dead0351aacc1acd2a643d2bc8cfe8eb414009

JJ commented on docs/eleccion-ci.md:
```diff
@@ -0,0 +1,94 @@
+* [**GitLab**](https://about.gitlab.com/): plataforma de desarrollo de ciclo de vida completo
+que incluye funciones de CI/CD. Al ser un proyecto en GitHub, su
+integración no es tan conveniente.
+* [**Bitbucket**](https://bitbucket.org/product/es/features/pipelines): servicio de desarrollo de software que incluye
```
Si no se puede usar desde GitHub, igual que arriba (y con Jenkins). Has puesto como criterio "compatibilidad con herramientas actuales", igual debería ser un requisito, y no criterio.

marinajcs replied:
Resuelto en https://github.com/marinajcs/asignacionTareas/pull/38/commits/b9dead0351aacc1acd2a643d2bc8cfe8eb414009

JJ reviewed (changes requested):
Está bastante bien con respecto a versiones y demás, pero en los criterios de elección debéis procurar afinar y poner sólo los que realmente sean posibles elecciones.
Siento no haberlo corregido antes; durante hoy y mañana si puedes cambiarlo lo corrijo sobre la marcha.

marinajcs committed:
Se afinan los criterios, fixes #37

Se incluyen como herramientas consideradas únicamente las que realmente podrían serlo, filtrando por el costo y la compatibilidad con las herramientas actuales, que pasa a ser un requisito principal.

JJ reviewed (approved):
:+1:
